lib/improver/nowcasting/optical_flow.py

```diff
@@ -321,23 +321,34 @@ def grid_spacing(coord):
         original_datetime, = \
             (cube.coord("time").units).num2date(cube.coord("time").points)
         new_datetime = original_datetime + timestep
-        new_time = (cube.coord("time").units).date2num(new_datetime)
 
-        if np.int64(new_time) == new_time:
-            new_time = np.int64(new_time)
-        else:
-            new_time = np.float64(new_time)
+        new_time = (cube.coord("time").units).date2num(new_datetime)
 
         advected_cube.coord("time").points = new_time
+        advected_cube.coord("time").convert_units(
+            "seconds since 1970-01-01 00:00:00")
+        advected_cube.coord("time").points = (
+            np.around(advected_cube.coord("time").points).astype(np.int64))
 
-        forecast_period_seconds = timestep.total_seconds()
+        try:
+            frt_coord = cube.coord("forecast_reference_time")
+            advected_cube.coord("forecast_reference_time").points = (
+                frt_coord.points)
+        except CoordinateNotFoundError:
+            frt_coord = cube.coord("time").copy()
+            frt_coord.rename("forecast_reference_time")
+            advected_cube.add_aux_coord(frt_coord)
```

Comment:
At the risk of being really annoying - this "try / except" block is doing stuff it doesn't need to do.  After the `try:` if `frt_coord` exists the second statement has no effect (the `advected_cube` is literally a copy of the original `cube`).  This is why I suggested the unit conversion as the first `try`.  There is then reassignment to the same variable `frt_coord` in line 344 below.  Please could you tidy up lines 333-346 into fewer lines without these reassignments.

Reply:
Is this what you're imagining?

```python
        try:
            advected_cube.coord("forecast_reference_time").convert_units(
                "seconds since 1970-01-01 00:00:00")
        except CoordinateNotFoundError:
            frt_coord = cube.coord("time").copy()
            frt_coord.rename("forecast_reference_time")
            advected_cube.add_aux_coord(frt_coord)
            advected_cube.coord("forecast_reference_time").convert_units(
                "seconds since 1970-01-01 00:00:00")

        frt_coord = advected_cube.coord("forecast_reference_time")
        advected_cube.coord("forecast_reference_time").points = (
            np.around(frt_coord.points).astype(np.int64))
```

Reply:
Pretty much, yes please.  If you can also get rid of the second assignment to `frt_coord` that would be ideal - I realise this makes a lot of line wrapping but assigning the same variable name to two different things generally isn't ideal.

Reply:
I've now modified this as suggested.